app/src/lib/stores/app-store.ts

```diff
@@ -5067,7 +5059,10 @@ export class AppStore extends TypedBaseStore<IAppState> {
       }`
     )
 
-    await this._refreshRepository(repository)
```

Comment:
@outofambit This particular removal of `_refreshRepository` makes me a little nervous. I'm not concerned with the changes you made in `checkoutBranch` since we'll always end up calling `refreshRepository` after the checkout has been completed and we can verify that during a checkout we don't need to refresh the repository state due to creating or popping a stash.

Here, on the other hand, we're calling this method (as far as I can tell) exclusively from the dispatcher as a result of the user clicking on the `restore` button. While looking through the difference in state updates between what `refreshRepository` does and what you're proposing I believe that we could run into problems since we're not updating the menu state here. It seems to me that if we're popping the last (or only) stash entry that should affect whether the `Show stash entry` menu item is enabled or not which would require updating the menu state.

Similarily I suspect that the enabledness of the `discard all changes` menu item would depend on whether the pop stash operation introduced new files to the working directory or not.

While we could certainly fix this particular problem I feel like it would be prudent to look at partial state refreshing a little more holistically to ensure that we don't run into similar problems in the future as we update the `refreshRepository` method to include other things.

We call `refreshRepository` every time the app is focused so I believe we should play it safe here and continue calling it in this instance.

Reply:
that's a great point! now that this method is decoupled from branch checkout, it's off of the slowest code path and we can definitely play it safe. thanks!

Reply:
0a65126